usecases/evaluate_scenario/evaluate_scenario.go

```diff
@@ -194,16 +186,16 @@ func EvalTestRunScenario(ctx context.Context,
 		),
 	)
 	defer span.End()
-	testRunIteration, err := repositories.EvalTestRunScenatioRepository.GetTestRunIterationByScenarioId(
+	testRunIteration, err := repositories.EvalTestRunScenarioRepository.GetTestRunIterationByScenarioId(
 		ctx, exec, params.Scenario.Id)
 	if err != nil {
 		return models.ScenarioExecution{}, errors.Wrap(err,
 			"error getting testrun scenario iteration in EvalTestRunScenario")
 	}
-	if testRunIteration.Id == "" {
-		return models.ScenarioExecution{}, nil
+	if testRunIteration == nil {
+		return models.ScenarioExecution{}, models.ErrNoTestRunFound
```

Comment:
ok, but considering how you're using this method in the decisions usecase, this will result in error logs all the time, no ? Since you decided to have all the logic of "checking if there even is a test run to execute" for a decision in this usecase, I think the "nothing to do" case should be handled differently.